distributed/versions.py

```diff
@@ -148,20 +148,24 @@ def error_message(scheduler, workers, client, client_name="client"):
         elif len(worker_versions) == 0:
             worker_versions = None
 
-        errs.append((pkg, client_version, scheduler_version, worker_versions))
+        level = "Critical" if pkg in ["python", "dask", "distributed"] else "Warning"
+
+        errs.append((pkg, client_version, scheduler_version, worker_versions, level))
         if pkg in notes_mismatch_package.keys():
             notes.append(f"-  {pkg}: {notes_mismatch_package[pkg]}")
 
     out = {"warning": "", "error": ""}
 
     if errs:
-        err_table = asciitable(["Package", client_name, "scheduler", "workers"], errs)
-        err_msg = f"Mismatched versions found\n\n{err_table}"
+        err_table = asciitable(
+            ["Package", client_name, "scheduler", "workers", "level"], errs
+        )
+        err_msg = f"Mismatched package versions found, this may cause instability. Please address critical mismatches before continuing.\n\n{err_table}"
```

Comment:
I could see a user being a bit confused by this message, since it doesn't really explain why they're critical. I like marking a severity level in the table, perhaps add a callout in the notes section below for mismatches in `dask`/`distributed`/`python` that adds some context? Something like:

```
# For python mismatch
note = "Variation in python versions can cause hard to debug serialization errors, we highly recommend using the same version of python everywhere"
```

Reply:
Good point thanks @jcrist. Added some notes.

Reply:
Thanks Jacob. I also think it would be good to revert the change here, or softening the language to something more like `"We recommend fixing any critical mismatches before continuing"` (and if you do that, perhaps only say that if there are critical mismatches?). I'll leave that up to your discretion.